gapic-common/lib/gapic/rest/error.rb

```diff
@@ -37,36 +50,85 @@ class << self
         # it tries to parse and set a detailed message and an error code from
         # from the Google Cloud's response body
         #
+        # @param err [Faraday::Error] the Faraday error to wrap
+        #
+        # @return [ Gapic::Rest::Error]
         def wrap_faraday_error err
           message = err.message
           status_code = err.response_status
+          status = nil
+          details = nil
+          headers = err.response_headers
 
           if err.response_body
-            msg, code = try_parse_from_body err.response_body
+            msg, code, status, details = try_parse_from_body err.response_body
             message = "An error has occurred when making a REST request: #{msg}" unless msg.nil?
             status_code = code unless code.nil?
           end
 
-          Gapic::Rest::Error.new message, status_code
+          Gapic::Rest::Error.new message, status_code, status: status, details: details, headers: headers
         end
 
         private
 
         ##
+        # @private
         # Tries to get the error information from the JSON bodies
         #
         # @param body_str [String]
         # @return [Array(String, String)]
```

Comment:
nit: this is now `Array(String,String,String,String)`

Reply:
:+1: